src/config/general.h

```diff
@@ -35,7 +35,7 @@ FILE_LICENCE ( GPL2_OR_LATER_OR_UBDL );
  */
 
 #define	NET_PROTO_IPV4		/* IPv4 protocol */
-#undef	NET_PROTO_IPV6		/* IPv6 protocol */
+//#define	NET_PROTO_IPV6		/* IPv6 protocol */
```

Comment:
Why is this changed to define and commented? Seems to me this should'nt be changed?

Reply:
IIRC because this file (`src/config/general.h`) is included ***after*** `src/config/defaults/efi.h`. Thus, if the `#undef NET_PROTO_IPV6` is left unchanged here, it ends up overrides the newly added `#define NET_PROTO_IPV6` from `src/config/defaults/efi.h`.

Changing it to `//#define` follows the `IMAGE_EFI` example:
https://github.com/ipxe/ipxe/blob/7656bf6a333a63e1da3483cf0b2a814252f56161/src/config/general.h#L113
https://github.com/ipxe/ipxe/blob/7656bf6a333a63e1da3483cf0b2a814252f56161/src/config/defaults/efi.h#L30

This logic accomplishes making the define in question default enabled for the EFI platform, while at the same time leaving it default disabled for all other platforms (unless explicitly disabled for some other specific platforms too).

Reply:
ok, Thanks for clarifying. I was worried this would enable it for all binary builds.